src/js/components/inspectors/MultiValueProperty.tsx

```diff
@@ -0,0 +1,178 @@
+  constructor(props) {
+    super(props);
+    this.state = { dragOver: 0 };
+  };
+
+  public handleChange(evt) {
+    const props = this.props;
+    const type = props.type;
+    let value = evt.target ? evt.target.value : evt;
+
+    // Ensure value is a number
+    if (type === 'number' || type === 'range') {
+      value = +value;
+    }
+
+    let newVal = props.processValue ? props.processValue(value, props) : value;
+
+    let arrayValues = [...props.value];
+    arrayValues.push(newVal);
+
+    evt.target.name = props.name;
+    props.onBlur ? props.onBlur(evt, arrayValues) : props.onKeyPress ? props.onKeyPress(evt, arrayValues) :  props.onChange(evt, arrayValues);
+  };
+
+  public handleUnBind(evt) {
+    const props = this.props;
+    let value = evt.target ? evt.target.dataset.value : evt;
+
+    const unbindVal = props.processValue ? props.processValue(value, props) : value;
+
+    let arrayValues = props.value.filter((val) => {
+      if (props.valueProperty) {
+        return val[props.valueProperty] !== unbindVal[props.valueProperty];
+      } else {
+        return val !== unbindVal;
+      }
+    });
+
+    evt.target.name = props.name;
+    props.onBlur ? props.onBlur(evt, arrayValues) : props.onKeyPress ? props.onKeyPress(evt, arrayValues) : props.onChange(evt, arrayValues);
+  };
+
+  public render() {
+    const props = this.props;
+    const label = props.label;
+    const valueProperty = props.valueProperty;
+    let dragOver = this.state.dragOver;
+    let labelEl = (<h3>{label}</h3>);
+    let selectedValsEl = [];
+    props.value.forEach( (item, index) => {
+      selectedValsEl.push(
+        <div className="property" key={index}>
+          <div className='label'>{index + 1}</div>
+          <div className='control'>
+            <div className='scale' data-value={valueProperty ? item[valueProperty] : item} onClick={(e) => this.handleUnBind(e)}>{props.subLabel || 'value'}</div>
+            <div className='field source' data-value={valueProperty ? item[valueProperty] : item}
+              onClick={(e) => this.handleUnBind(e)}>{valueProperty ? item[valueProperty] : item}</div>
+          </div>
+        </div>
+      );
+    });
+    
+    const className = 'property' +
+      (props.droppable && props.dragging ? ' droppable' : '') +
+      (props.droppable && dragOver ? ' drag-over' : '') +
+      (props.firstChild ? ' first-child' : '');
+
+    return (
+      <div>
+        {labelEl}
+        <div className={className}
+          onDragEnter={() => this.setState({ dragOver: ++dragOver })}
+          onDragLeave={() => this.setState({ dragOver: --dragOver })}>
+          {selectedValsEl}
+          <Property {...props} label="Add" value={null} onChange={props.onChange ? (e) => this.handleChange(e) : null} onBlur={props.onBlur ? (e) => this.handleChange(e) : null} onKeyPress={props.onKeyPress ? (e) => this.handleChange(e) : null} />
```

Comment:
why `onKeyPress`? did the existing `onChange` handling in `Property` not work?

Reply:
so the issue with onchange is that it fires for every change(even while you are typing) which we don't want for the manual text input. so just adding support for keypress so that we can listen for the enter key.